.circleci/config.yml

```diff
@@ -1,43 +1,152 @@
+      - run:
+          name: "Prepare Checksum for Gradle Dependencies"
+          command:  find . -maxdepth 3 -type f \( -name "build.gradle" -o -name "gradle-wrapper.properties" \) -exec shasum {} + | sort | tee gradle-build-files.checksum
       - restore_cache:
+          name: "Restore Gradle Cache"
           keys:
-          - v1-dependencies-{{ checksum "pom.xml" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
-
-      - run: mvn dependency:go-offline
-
+            - gradle-v3-{{ checksum "gradle-build-files.checksum" }}
+            - gradle-v3-
+  save_gradle_cache:
+    description: "Saves the gradle cache. Requires the checksum file created in the restore_gradle_cache command"
+    steps:
       - save_cache:
+          name: "Save Gradle Dependencies Cache"
+          key: gradle-v3-{{ checksum "gradle-build-files.checksum" }}
           paths:
-            - ~/.m2
-          key: v1-dependencies-{{ checksum "pom.xml" }}
-        
-      # run tests!
-      - run: mvn integration-test test
+            - "~/.gradle"
+  prepare_build_environment:
+    description: "Prepares the build environment and installs java."
+    steps:
+      - run:
+          name: "Prepare build environment"
+          command: |
+            set -e
+            # sudo update-alternatives --set java /usr/lib/jvm/java-11-openjdk-amd64/jre/bin/java
+            # sudo update-alternatives --set javac /usr/lib/jvm/java-11-openjdk-amd64/bin/javac
+            # echo "export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64" >> $BASH_ENV
+            java -version
+            javac -version
+  publish_artifacts:
+    description: "Publishes the gradle build artifacts. Will retry two times if artifactory is not responsive with an increasing sleep time."
+    steps:
+      - run:
+          name: "Resilient Publish cplace Artifacts"
+          command: |
+            set -e
+            _retry_count=0
+            while true; do
+              if [[ "$_retry_count" -gt 2 ]]; then
+                echo "Failed to upload the artifacts after $_retry_count retries!"
+                exit 1
+              fi
+              if ./gradlew --stacktrace publish; then
+                echo "Successfully uploaded the artifacts."
+                break
+              fi
+              _retry_count=${_retry_count + 1}
+              sleep ${_retry_count * 5}
+            done
+jobs:
+  build-and-test:
+    machine:
+      image: ubuntu-2004:202010-01
+    parallelism: 1
+    shell: /bin/bash --login
+    environment:
+      GRADLE_OPTS: "-Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2"
+    parameters:
+      PUBLISH_SNAPSHOT:
+        type: boolean
+        default: false
+    steps:
+      - cached_checkout
+      #      - shallow_checkout
+      - restore_gradle_cache
+      - prepare_build_environment
+      - run:
+          name: "Build and Test"
+          command: ./gradlew --continue --stacktrace build
```

Comment:
This is a Maven project, Gradle won't work here.

Compare the steps `mvn dependency:go-offline` and `mvn integration-test test` from the old file.